homeassistant/components/rituals_perfume_genie/sensor.py

```diff
@@ -12,6 +21,57 @@
 from .entity import DiffuserEntity
 
 
+@dataclass
+class RitualsEntityDescriptionMixin:
+    """Mixin values for Rituals entities."""
+
+    value_fn: Callable[[Diffuser], int | str]
+
+
+@dataclass
+class RitualsSensorEntityDescription(
+    SensorEntityDescription, RitualsEntityDescriptionMixin
+):
+    """Class describing Rituals sensor entities."""
+
+    has_fn: Callable[[Diffuser], bool] = lambda _: True
+
+
+ENTITY_DESCRIPTIONS = (
+    RitualsSensorEntityDescription(
+        key="battery_percentage",
+        name="Battery",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda diffuser: diffuser.battery_percentage,
+        has_fn=lambda diffuser: diffuser.has_battery,
+    ),
+    RitualsSensorEntityDescription(
+        key="fill",
+        name="Fill",
+        icon="mdi:beaker",
+        entity_category=EntityCategory.DIAGNOSTIC,
```

Comment:
So icon is now static, but did you mean to make this a diagnostic entity?

Reply:
Yes and yes :)